langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/InternalAzureOpenAiHelper.java

```diff
@@ -120,18 +120,41 @@ public static List<com.azure.ai.openai.models.ChatRequestMessage> toOpenAiMessag
 
     public static com.azure.ai.openai.models.ChatRequestMessage toOpenAiMessage(ChatMessage message) {
         if (message instanceof AiMessage) {
-            ChatRequestAssistantMessage chatRequestAssistantMessage = new ChatRequestAssistantMessage(getOrDefault(message.text(), ""));
+            AiMessage aiMessage = (AiMessage) message;
+            ChatRequestAssistantMessage chatRequestAssistantMessage = new ChatRequestAssistantMessage(getOrDefault(aiMessage.text(), ""));
             chatRequestAssistantMessage.setFunctionCall(functionCallFrom(message));
             return chatRequestAssistantMessage;
         } else if (message instanceof ToolExecutionResultMessage) {
             ToolExecutionResultMessage toolExecutionResultMessage = (ToolExecutionResultMessage) message;
             return new ChatRequestFunctionMessage(nameFrom(message), toolExecutionResultMessage.text());
         } else if (message instanceof SystemMessage) {
-            return new ChatRequestSystemMessage(message.text());
-        } else {
-            ChatRequestUserMessage chatRequestUserMessage = new ChatRequestUserMessage(message.text());
+            SystemMessage systemMessage = (SystemMessage) message;
+            return new ChatRequestSystemMessage(systemMessage.text());
+        } else if (message instanceof UserMessage) {
+            UserMessage userMessage = (UserMessage) message;
+            ChatRequestUserMessage chatRequestUserMessage;
+            if (userMessage.hasSingleText()) {
+                chatRequestUserMessage = new ChatRequestUserMessage(((TextContent) userMessage.contents().get(0)).text());
+            } else {
+                chatRequestUserMessage = new ChatRequestUserMessage(userMessage.contents().stream()
+                        .map(content -> {
+                            if (content instanceof TextContent) {
+                                String text = ((TextContent) content).text();
+                                return new ChatMessageTextContentItem(text);
+                            } else if (content instanceof ImageContent) {
+                                ImageContent imageContent = (ImageContent) content;
+                                ChatMessageImageUrl imageUrl = new ChatMessageImageUrl(imageContent.image().url().toString());
```

Comment:
`imageContent.image().url()` can return `null` in case image is provided as `base64Data`, see `ImageContent` constructors for all possible combinations

Reply:
Indeed! But as we can't send a base64 string (the `ChatMessageImageContentItem` can only support URLs), what do we do in case of a base64 data, do we drop it?

Reply:
Maybe throw an exception in such case? Dropping it silently will be very confusing for the user